### PR TITLE
Deflake informer and ovn/pods tests

### PR DESCRIPTION
**Pull request by aojea**
**- What this PR does and why is it needed**

Some tests depends that an event is processed by the handler, and some times this takes longer than expected failing the test.
We can poll periodically during a small amount of time to avoid this.

**- Special notes for reviewers**

**- How to verify it**
Test job pass

**- Description for the changelog**

### CODE REVIEW

**Comment by dcbw** [comment]
@aojea can we do the 500ms polling thing from a function instead? That's a lot of code dupe.

**Comment by dcbw** [comment]
Eg something like

`Expect(pollForSomething()).NotTo(HaveOccurred())`

**Comment by aojea** [comment]
updated with the suggestions
PTAL

File: go-controller/pkg/ovn/pods_test.go [hunk 1]
```diff
@@ -231,12 +239,7 @@ var _ = Describe("OVN Pod Operations", func() {
 				Expect(err).NotTo(HaveOccurred())
 				Eventually(fExec.CalledMatchesExpected).Should(BeTrue(), fExec.ErrorDesc)
 
-				pod, err = fakeOvn.fakeClient.CoreV1().Pods(t.namespace).Get(context.TODO(), t.podName, metav1.GetOptions{})
-				Expect(err).NotTo(HaveOccurred())
-
-				podAnnotation, ok := pod.Annotations[util.OvnPodAnnotationName]
-				Expect(ok).To(BeTrue())
-				Expect(podAnnotation).To(MatchJSON(`{"default": {"ip_addresses":["` + t.podIP + `/24"], "mac_address":"` + t.podMAC + `", "gateway_ips": ["` + t.nodeGWIP + `"], "ip_address":"` + t.podIP + `/24", "gateway_ip": "` + t.nodeGWIP + `"}}`))
+				Eventually(getPodAnnotations(fakeOvn.fakeClient, t.namespace, t.podName), 2).Should(MatchJSON(`{"default": {"ip_addresses":["` + t.podIP + `/24"], "mac_address":"` + t.podMAC + `", "gateway_ips": ["` + t.nodeGWIP + `"], "ip_address":"` + t.podIP + `/24", "gateway_ip": "` + t.nodeGWIP + `"}}`))
```
Comment (danwinship) [inline go-controller/pkg/ovn/pods_test.go:242]: This has the same problem as in the other bug; you are passing a constant value to `Eventually`, not passing a function for it to run repeatedly, so it is still just requiring the annotation to be set immediately.

Comment (dcbw) [reply go-controller/pkg/ovn/pods_test.go:242]: Gah you're right, it needs to be:

```
Eventually(func() string { return getPodAnnotations(fakeOvn.fakeClient, t.namespace, t.podName) }, 2).Should(MatchJSON(...))
```

Comment (aojea) [reply go-controller/pkg/ovn/pods_test.go:242]: :facepalm:

**Comment by danwinship** [comment]
lgtm

(the other possibility would be to rename `getPodAnnotations` to something like `podAnnotationGetter` and make it be a function that returns a function that gets the annotation)

**Pull request merged**